model/publishing/delivery/PublishingDeliveryService.php

```diff
@@ -121,6 +118,25 @@ public function getSyncFields()
         return $deliveryFieldsOptions;
     }
 
+    /**
+     * @param $delivery
+     * @return QueueDispatcher
+     */
+    public function getDispatcher($delivery)
+    {
+        /** @var QueueDispatcher $queueDispatcher */
+        $queueDispatcher = $this->getServiceManager()->get(QueueDispatcher::SERVICE_ID);
+        if ($taskResource = $queueDispatcher->getTaskResource($delivery) ) {
```

Comment:
FYI, getTaskResource() will be deprecated in the UI component branch (which is under review and will be merged soon) and this method won't be used at all for delivery compilation, so it might be a good idea to figure out a different logic now.